DataSetExplorer/DataSetBuilder/CaDETToDataSetBuilder.cs

```diff
@@ -115,8 +115,12 @@ private int DetermineNumberOfInstances(int totalNumber)
 
         private List<DataSetInstance> CaDETToDataSetClasses(List<CaDETClass> cadetClasses)
         {
-            return cadetClasses.Select(c => 
-                new DataSetInstance(c.FullName, GetCodeUrl(c.FullName), _projectAndCommitUrl, SnippetType.Class)).ToList();
+            return cadetClasses.Select(c =>
+            {
+                var dataSetInstance = new DataSetInstance(c.FullName, GetCodeUrl(c.FullName), _projectAndCommitUrl, SnippetType.Class);
```

Comment:
Could we avoid the anonymous function with an expanded constructor?

Reply:
Do you think we should remove the select method? As far as I understand, we need it to transform all classes to data set instances. Maybe I didn't understand the comment because I'm not completely familiar with C#. :) Do you mean:

return cadetClasses.Select(c =>
    {
        return new DataSetInstance(c.FullName, GetCodeUrl(c.FullName), _projectAndCommitUrl, SnippetType.Class)
        {
            MetricFeatures = _cadetProject.GetMetricsForCodeSnippet(c.FullName)
        };
    }).ToList();

Reply:
I was thinking more along the lines of:

    return cadetClasses.Select(c => return new DataSetInstance(
            c.FullName, GetCodeUrl(c.FullName), _projectAndCommitUrl, SnippetType.Class, _cadetProject.GetMetricsForCodeSnippet(c.FullName)
        ).ToList()